src/Spice86.Core/Emulator/OperatingSystem/DosPathResolver.cs

```diff
@@ -0,0 +1,236 @@
+namespace Spice86.Core.Emulator.OperatingSystem;
+
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+using Spice86.Core.Emulator.OperatingSystem.Structures;
+using Spice86.Shared.Emulator.Errors;
+using Spice86.Shared.Interfaces;
+using Spice86.Shared.Utils;
+
+/// <inheritdoc />
+public class DosPathResolver : IDosPathResolver {
+
+    /// <inheritdoc />
+    public IDictionary<char, MountedFolder> DriveMap { get; private set; } = new Dictionary<char, MountedFolder>();
+
+    /// <inheritdoc />
+    public char CurrentDrive { get; private set; }
+
+    /// <inheritdoc />
+    public string CurrentHostDirectory {
```

Comment:
Non-nullable property is not set after invoking constructor

Reply:
This is not reported by the compiler, as this is a calculated property.

Reply:
I meant that you should make it nullable if you are not going to give it a default value. That way code that uses the interface can see that this property may not have a valid value yet.

Reply:
I consider null to be a code smell. Null questions in any code base make it less readable. I'd really like to avoid it if possible.

Reply:
Then make sure to initialize these properties from the constructor.

Reply:
Done.